packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx

```diff
@@ -38,34 +38,53 @@ export class MarkdownPartRenderer implements ChatResponsePartRenderer<MarkdownCh
         }
         return -1;
     }
-    private renderMarkdown(md: MarkdownString): HTMLElement {
-        return this.renderer.render(md).element;
-    }
     render(response: MarkdownChatResponseContent | InformationalChatResponseContent): ReactNode {
         // TODO let the user configure whether they want to see informational content
         if (InformationalChatResponseContent.is(response)) {
             // null is valid in React
             // eslint-disable-next-line no-null/no-null
             return null;
         }
-        return <MarkdownWrapper data={response.content} renderCallback={this.renderMarkdown.bind(this)}></MarkdownWrapper>;
+
+        // return <div ref={ref}></div>;
+        return <MarkdownRender response={response} />;
     }
 
 }
 
-export const MarkdownWrapper = (props: { data: MarkdownString, renderCallback: (md: MarkdownString) => HTMLElement }) => {
+const MarkdownRender = ({ response }: { response: MarkdownChatResponseContent | InformationalChatResponseContent }) => {
     // eslint-disable-next-line no-null/no-null
-    const ref: React.MutableRefObject<HTMLDivElement | null> = useRef(null);
+    const ref = useMarkdownRendering(response.content.value);
 
-    useEffect(() => {
-        const myDomElement = props.renderCallback(props.data);
+    return <div ref={ref}></div>;
+};
 
+/**
+ * This hook uses markdown-it directly to render markdown.
+ * The reason to use markdown-it directly is that the MarkdownRenderer is
+ * overriden by theia with a monaco version. This monaco version strips all html
+ * tags from the markdown with empty content.
+ * This leads to unexpected behavior when rendering markdown with html tags.
+ *
+ * @param markdown the string to render as markdown
+ * @returns the ref to use in an element to render the markdown
+ */
+export const useMarkdownRendering = (markdown: string) => {
+    // eslint-disable-next-line no-null/no-null
+    const ref = useRef<HTMLDivElement | null>(null);
```

Comment:
Is there a benefit of using `null` over `undefined`?

Reply:
the ref paramerter wants null, can't be used with undefined